packages/common/src/authentication/api/hooks/useStores.ts

```diff
@@ -0,0 +1,13 @@
+import { useMutation } from 'react-query';
+import { useQueryParams } from '@common/hooks';
+import { useAuthApi } from './useAuthApi';
+
+export const useStores = () => {
+  const api = useAuthApi();
+  const initialListParameters = { initialSortBy: { key: 'code' } };
+  const { filterBy, first, offset } = useQueryParams(initialListParameters);
+  const getStores = api.get.stores({ filterBy, first, offset });
+  const { mutate, ...rest } = useMutation(getStores);
```

Comment:
You can explain this to me in person next week if that's easier, but I'm just wondering why this is a mutation?

Reply:
using it so that I could called `mutateAsync` and `await` the response, and so that I could call explicitly.

Mutations allow you to call them when required, as opposed to queries which are called when you call `useQuery`, so it would run when the page loads. In this case, the request only needs to be called if there is no MRU, in most cases it is not required.

You can prevent / delay execution with a query by setting `enabled`, so in this case I could call 

``` useQuery(getStores, { enabled: !mostRecentlyUsedCredentials?.store }) ```

I thought that a mutation is clearer, and it is also helpful to be able to await. Given how quick the query is, there's little need to prefetch the results